x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/cluster/metadata/MetadataMigrateToDataTiersRoutingService.java

```diff
@@ -334,6 +334,17 @@ private static LifecyclePolicy migrateSingleILMPolicy(String nodeAttrName, Lifec
                         phase.getName());
                 }
 
+                // we removed the allocate action allocation rules (or the action completely) so let's check if there is an
+                // explicit migrate action that's disabled, and enable it
+                if (actionMap.containsKey(MigrateAction.NAME)) {
+                    MigrateAction migrateAction = (MigrateAction) actionMap.get(MigrateAction.NAME);
+                    if (migrateAction.isEnabled() == false) {
+                        MigrateAction enabledMigrateAction = new MigrateAction(true);
+                        actionMap.put(MigrateAction.NAME, enabledMigrateAction);
+                        logger.debug("ILM policy [{}], phase [{}]: enabled the migrate action", lifecyclePolicy.getName(), phase.getName());
+                    }
+                }
```

Comment:
We don't necessarily have to explicitly add the migrate step, we could rather just do `actionMap.remove(MigrateAction.NAME)` since the default is enabled when no allocation step is configured?